src/main/scala/io/iohk/ethereum/db/dataSource/DataSource.scala

```diff
@@ -0,0 +1,26 @@
+package io.iohk.ethereum.db.dataSource
+
+trait DataSource {
+
+  type Key = Array[Byte]
+  type Value = Array[Byte]
+
+  /**
+    * This function obtains the associated value to a key, if there exists one.
+    *
+    * @param namespace which will be searched for the key.
```

Comment:
So, if I am getting this design right then I'd add a new `type Namespace = IndexedSeq[Byte]` and make all changes in code resulting from that.

If it's going to take you more than one hour, then let's keep it as it and move forward :)